pkg/operator/minio/controller.go

```diff
@@ -0,0 +1,301 @@
+			VolumeClaimTemplates: []v1.PersistentVolumeClaim{
+				{
+					ObjectMeta: meta_v1.ObjectMeta{
+						Name:      minioVolumeName,
+						Namespace: namespace,
+					},
+					Spec: v1.PersistentVolumeClaimSpec{
+						AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+						Resources: v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceStorage: resource.MustParse(spec.StorageSize),
+							},
+						},
+					},
+				},
+			},
+			ServiceName: name,
+			// TODO: liveness probe
+		},
+	}
+
+	return appsClient.StatefulSets(namespace).Create(&ss)
+}
+
+func (c *MinioController) makeMinioService(name, namespace string, spec miniov1alpha1.ObjectStoreSpec, ownerRef meta_v1.OwnerReference) (*v1.Service, error) {
+	coreV1Client := c.context.Clientset.CoreV1()
+
+	svc, err := coreV1Client.Services(namespace).Create(&v1.Service{
+		ObjectMeta: meta_v1.ObjectMeta{
+			Name:            name,
+			Namespace:       namespace,
+			Labels:          map[string]string{"app": minioLabel},
+			OwnerReferences: []meta_v1.OwnerReference{ownerRef},
+		},
+		Spec: v1.ServiceSpec{
+			Type:     v1.ServiceTypeNodePort,
+			Selector: map[string]string{"app": minioLabel},
+			Ports:    []v1.ServicePort{{Port: spec.Port}},
+		},
+	})
+
+	return svc, err
+}
+
+func (c *MinioController) onAdd(obj interface{}) {
+	objectstore := obj.(*miniov1alpha1.ObjectStore).DeepCopy()
+
+	ownerRef := meta_v1.OwnerReference{
+		APIVersion: ObjectStoreResource.Version,
+		Kind:       ObjectStoreResource.Kind,
+		Name:       objectstore.Namespace,
+		UID:        types.UID(objectstore.ObjectMeta.UID),
+	}
+
+	// Create the headless service.
+	logger.Infof("Creating Minio headless service %s.", objectstore.Name)
+	_, err := c.makeMinioHeadlessService(objectstore.Name, objectstore.Namespace, objectstore.Spec, ownerRef)
+	if err != nil {
+		logger.Errorf("failed to create minio headless service: %v", err)
+		return
+	}
+	logger.Infof("Finished creating Minio headless service %s.", objectstore.Name)
+
+	// Create the stateful set.
+	logger.Infof("Creating Minio stateful set %s.", objectstore.Name)
+	_, err = c.makeMinioStatefulSet(objectstore.Name, objectstore.Namespace, objectstore.Spec, ownerRef)
+	if err != nil {
+		logger.Errorf("failed to create minio stateful set: %v", err)
+		return
+	}
+	logger.Infof("Finished creating Minio stateful set %s.", objectstore.Name)
+
+	// Create the nodeport service.
+	svcName := objectstore.Name + "-service"
+	logger.Infof("Creating Minio service %s.", svcName)
+	_, err = c.makeMinioService(svcName, objectstore.Namespace, objectstore.Spec, ownerRef)
+	if err != nil {
+		logger.Errorf("failed to create minio nodeport service: %v", err)
+		return
+	}
+	logger.Infof("Finished creating Minio service %s.", svcName)
+}
+
+func (c *MinioController) onUpdate(oldObj, newObj interface{}) {
+	oldStore := oldObj.(*miniov1alpha1.ObjectStore).DeepCopy()
+	newStore := newObj.(*miniov1alpha1.ObjectStore).DeepCopy()
+
+	_ = oldStore
```

Comment:
maybe log here that the update event was received but it is not supported yet so it was ignored?